lib/rules/no-constant-condition.js

```diff
@@ -115,15 +118,74 @@ module.exports = {
         }
 
         /**
-         * Reports when the given node contains a constant condition.
+         * Tracks when the given node contains a constant condition.
          * @param {ASTNode} node The AST node to check.
          * @returns {void}
          * @private
          */
         function checkConstantCondition(node) {
```

Comment:
Can you rename this function to something like `trackConstantConditionLoop`? There are several functions in this file that do different things related to checking constant conditions, so the current name is a bit confusing.